src/anndata/_io/specs/lazy_methods.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from pathlib import Path, PurePosixPath
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, Literal, overload
+
+import h5py
+import numpy as np
+from scipy import sparse
+
+import anndata as ad
+from anndata.compat import H5Array, H5Group, ZarrArray, ZarrGroup
+
+from .registry import _LAZY_REGISTRY, IOSpec
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+
+
+@overload
+def make_index(
+    *,
+    is_csc: Literal[True],
+    stride: int,
+    shape: tuple[int, int],
+    block_id: tuple[int, int],
+) -> tuple[slice, slice]: ...
+@overload
+def make_index(
+    *,
+    is_csc: Literal[False],
+    stride: int,
+    shape: tuple[int, int],
+    block_id: tuple[int, int],
+) -> tuple[slice]: ...
+
+
+def make_index(
```

Comment:
I would rename this. The name sounds like it is building and index as in an index for speeding up look ups. But what the function is doing is making an object that can be used to index in to another array. 

Maybe `make_block_indexer`? Or `_block_indexer`?